string state
string phase
uint32 pickup_index
uint32 pickup_total
string last_error
